src/components/Dialogs/RestFreqDialog/RestFreqDialogComponent.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from "react";
+import classNames from "classnames";
+import {observer} from "mobx-react";
+import {action, observable, computed, makeObservable, autorun} from "mobx";
+import {FormGroup, IDialogProps, Button, Intent, Classes, Text, HTMLSelect, NonIdealState} from "@blueprintjs/core";
+import {DraggableDialogComponent} from "components/Dialogs";
+import {ClearableNumericInputComponent} from "components/Shared";
+import {AppStore, HelpType} from "stores";
+import {FrequencyUnit} from "models";
+import "./RestFreqDialogComponent.scss";
+
+@observer
+export class RestFreqDialogComponent extends React.Component {
+    private defaultRestFreq: number;
+    private defaultUnit: FrequencyUnit = FrequencyUnit.HZ;
+
+    @observable filename: string = "";
+    @observable private restFreq: number;
+    @observable private unit: FrequencyUnit = FrequencyUnit.HZ;
+
+    @action private setFilename = (val: string) => {
+        this.filename = val;
+    };
+
+    @action private setRestFreq = (val: number) => {
+        this.restFreq = val;
+    };
+
+    @action private setUnit = (val: FrequencyUnit) => {
+        this.unit = val;
+    };
+
+    @action private restoreDefaults = () => {
+        this.restFreq = this.defaultRestFreq;
+        this.unit = this.defaultUnit;
+    };
+
+    constructor(props: any) {
+        super(props);
+        makeObservable(this);
+
+        autorun(() => {
+            const appStore = AppStore.Instance;
+            const frame = appStore.getFrame(appStore.dialogStore.restFreqDialogFileId);
+            this.setFilename(frame?.filename);
+
+            const headerRestFreq = this.convertUnit(frame?.headerRestFreq);
+            this.defaultRestFreq = headerRestFreq?.value;
+            this.defaultUnit = headerRestFreq?.unit;
+
+            const customRestFreq = this.convertUnit(frame?.customRestFreq);
+            this.setRestFreq(customRestFreq?.value);
+            this.setUnit(customRestFreq?.unit);
+        });
+    }
+
+    @computed private get inValidInput(): boolean {
+        return !isFinite(this.restFreq);
+    }
+
+    @computed private get defaultInfo(): string {
+        return isFinite(this.defaultRestFreq) ? `Header: ${this.defaultRestFreq} ${this.defaultUnit}` : undefined;
+    }
+
+    private convertUnit = (restFreq: number) => {
+        if (restFreq >= 1e9) {
+            return {value: restFreq / 1e9, unit: FrequencyUnit.GHZ};
+        } else if (restFreq >= 1e6) {
+            return {value: restFreq / 1e6, unit: FrequencyUnit.MHZ};
+        } else if (restFreq >= 1e3) {
+            return {value: restFreq / 1e3, unit: FrequencyUnit.KHZ};
+        } else {
+            return {value: restFreq, unit: FrequencyUnit.HZ};
+        }
+    };
+
+    private saveRestFreq = () => {
```

Comment:
perhaps make this an `@action` to bundle the called actions together

Reply:
I'm not sure if it's necessary because only `hideRestFreqDialog()` is an `@action` function. or do we consider also add `@action` to `updateCustomRestFreq()`?

Reply:
yes, that should also be an `@action`, since it calls multiple actions. Functions that call multiple actions should also be actions, so that we reduce the number of reactions

Reply:
@veggiesaurus added `@action` to `saveRestFreq()` and `updateCustomRestFreq()`. The features work fine after the change